### PR TITLE
feat: review lock compute units on state manager openrouter

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: atoma-state/src/state_manager.rs [hunk 1]
```diff
@@ -1589,6 +1590,7 @@ impl AtomaState {
         .bind(total_compute_units)
         .bind(stack_small_id)
         .bind(ratio_for_claim_stacks)
+        .bind(concurrent_requests)
```
[the remark this answers is not in the document]
Comment (jorgeantonio21) [reply atoma-state/src/state_manager.rs:1593]: Wth ? How come there are only 4 bound values ? I don't get it @maschad

File: atoma-state/src/state_manager.rs [hunk 2]
```diff
@@ -5380,12 +5382,55 @@ mod tests {
...
+            is_claimed: false,
+            is_locked_for_claim: false,
+        }).await?;
+        // Test case 6: Edge case - estimated equals num_compute_units, but with concurrent requests
+        let UpdateStackNumComputeUnitsAndClaimFunds {
+            ratio,
+            stack_computed_units,
+            is_confidential,
+        } = state
+            .update_stack_num_compute_units(2, 50, 50, 0.95, 1)
```
Comment (maschad) [inline atoma-state/src/state_manager.rs:5416]: the `stack_small_id` here is first parameter, but in the SQL query it's bound to `$3`.  You can either re-order the bindings or modify the SQL query to match the current binding order as suggested [here](https://github.com/atoma-network/atoma-node/pull/541/files/98d1aec7c8dc86e3304d8a7b51ce3a8bfc32b7b2#r2029310325)

**Review by maschad — CHANGES_REQUESTED** [review]
Looks good @jorgeantonio21 , left some comments

File: atoma-state/src/state_manager.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorgeantonio21) [reply atoma-state/src/state_manager.rs:1593]: That's not true

File: atoma-state/src/state_manager.rs (hunk 2 above)
Comment (jorgeantonio21) [reply atoma-state/src/state_manager.rs:5416]: Address the comment.

File: atoma-state/src/state_manager.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (maschad) [reply atoma-state/src/state_manager.rs:1593]: My bad, the diff had obscured line 1589 from me. Apologies

**Review by maschad — APPROVED** [review]
